src/app/core/services/user/current-user.service.ts

```diff
@@ -1,28 +1,55 @@
 import { Injectable } from '@angular/core';
-import { AngularFireAuth } from '@angular/fire/compat/auth';
-import { catchError, map, of, shareReplay, switchMap } from 'rxjs';
+import {
+	BehaviorSubject,
+	map,
+	Observable,
+	of,
+	shareReplay,
+	switchMap,
+	tap
+} from 'rxjs';
 import { IUser } from '@core/models/User';
 import { UserFirebaseService } from '../firebase/firebase-entities/userFirebase.service';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
 
 @Injectable({
 	providedIn: 'root'
 })
 export class CurrentUserService {
-	public currentUser$ = this.afAuth.user.pipe(
-		switchMap(user =>
-			user
-				? this.userFireStore
-						.getUserDataByEmail(user.email)
-						.pipe(catchError(() => of({ roleId: '' } as IUser)))
-				: of({ roleId: '' } as IUser)
-		),
-		shareReplay(1)
-	);
-
-	public isUserLogin$ = this.currentUser$.pipe(map(user => user.roleId !== ''));
+	public currentUser$: Observable<IUser>;
+	public isUserLogin$: Observable<boolean>;
+	public isAdmin: BehaviorSubject<boolean>;
+	public isDatabaseInitialized: BehaviorSubject<boolean>;
 
 	constructor(
 		private afAuth: AngularFireAuth,
 		private userFireStore: UserFirebaseService
-	) {}
+	) {
+		this.isAdmin = new BehaviorSubject<boolean>(false);
+		this.isDatabaseInitialized = new BehaviorSubject<boolean>(true);
+		this.initUser();
+	}
+
+	initUser(): void {
+		this.currentUser$ = this.afAuth.user.pipe(
+			switchMap(user =>
+				user
+					? this.userFireStore.getUserData(user.uid)
+					: of({ roleId: '' } as IUser)
```

Comment:
Need to check!

Reply:
Return your changes, with fetching by email